src/main/java/seedu/momentum/model/project/Deadline.java

```diff
@@ -33,49 +42,94 @@ public Deadline() {
     /**
      * Constructs a {@code Deadline}.
      *
-     * @param date A valid date.
+     * @param date        A valid date after or on created date.
+     * @param createdDate A created date.
      */
-    public Deadline(String date) {
+    public Deadline(String date, Date createdDate) {
         requireNonNull(date);
+        requireNonNull(createdDate);
         checkArgument(Date.isValid(date), Date.MESSAGE_CONSTRAINTS);
+        checkArgument(isBeforeCreatedDate(date, createdDate), CREATED_DATE_MESSAGE_CONSTRAINT);
         this.date = Optional.of(new Date(date));
         this.time = Optional.empty();
     }
 
     /**
      * Constructs a {@code Deadline}.
      *
-     * @param date A valid date.
-     * @param time A valid time.
+     * @param date        A valid date after or on created date.
+     * @param time        A valid time.
+     * @param createdDate A created date.
      */
-    public Deadline(String date, String time) {
+    public Deadline(String date, String time, Date createdDate) {
         requireNonNull(date, time);
+        requireNonNull(createdDate);
         checkArgument(Date.isValid(date), Date.MESSAGE_CONSTRAINTS);
         checkArgument(Time.isValid(time), Time.MESSAGE_CONSTRAINTS);
+        checkArgument(isBeforeCreatedDate(date, createdDate), CREATED_DATE_MESSAGE_CONSTRAINT);
```

Comment:
Do the others think it would be a problem that the time for deadline is not checked against created time? I know we are not saving the created time and this would mean we would need to make changes to the created date to include time but I feel like if we are checking the date, we should be checking the time as well. What do the others think?

Reply:
I think the main reason to include check for created date is to ensure that we will never run into a utc offset problem, rather than making sure the deadline is actually after the created date and we are also not currently storing the created time. So I think its fine since only the created date is stored as long as the date of the deadline is later than created date we can just take it as valid.

Reply:
Alright, sure! I'm okay with not checking the time